src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -92,7 +100,28 @@ public boolean addAll(final Collection<? extends T> collection) {
     @Override
     public boolean addAll(final int index,
             final Collection<? extends T> collection) {
-        throw new UnsupportedOperationException();
+        this.checkCollectionNotNull(collection);
+        this.checkIndexToAdd(index);
+        int startIndex = index;
+        int collectionSize = collection.size();
+
+        while (this.currentIndex + collectionSize > this.data.length - 1) {
+            this.increaseSize();
```

Comment:
Imagine you have collection of 100 and array of 10. You will increase the size from 10 to 20 then 20 to 40 then 40 to 80 then 80 to 160.

You will book 5 times and move the elements 5 times.
Can you do it in one time?
Do you like this way ?! Do you think it is efficient?

Reply:
it is better to be done one time